packages/frontend/src/features/scheduler/components/SchedulerModalContent2.tsx

```diff
@@ -0,0 +1,139 @@
+import { DialogProps } from '@blueprintjs/core';
+import {
+    ApiError,
+    CreateSchedulerAndTargetsWithoutIds,
+    SchedulerAndTargets,
+} from '@lightdash/common';
+import { Box, Button, Group } from '@mantine/core';
+import { FC, useEffect, useState } from 'react';
+import {
+    UseMutationResult,
+    UseQueryResult,
+} from 'react-query/types/react/types';
+import { useHistory, useLocation } from 'react-router-dom';
+import { getSchedulerUuidFromUrlParams } from '../utils';
+import SchedulersList from './SchedulersList';
+
+import {
+    CreateStateContent,
+    UpdateStateContent,
+} from './SchedulerModalContent';
+
+enum States {
+    LIST,
+    CREATE,
+    EDIT,
+}
+
+const ListStateContent: FC<{
+    schedulersQuery: UseQueryResult<SchedulerAndTargets[], ApiError>;
+    onClose: () => void;
+    onConfirm: () => void;
+    onEdit: (schedulerUuid: string) => void;
+}> = ({ schedulersQuery, onClose, onConfirm, onEdit }) => {
+    return (
+        <>
+            <Box py="sm" mih={220}>
+                <SchedulersList
+                    schedulersQuery={schedulersQuery}
+                    onEdit={onEdit}
+                />
+            </Box>
+            <Group
+                spacing="xs"
+                position="right"
+                // TODO: this css is basically a sticky-footer setup for Mantine modals.
+                // It should be shared somewhere else.
+                sx={(theme) => ({
+                    position: 'sticky',
+                    backgroundColor: 'white',
+                    border: `1px solid ${theme.colors.gray[4]}`,
+                    bottom: 0,
+                    margin: `-${theme.spacing.md}`,
+                    padding: theme.spacing.md,
+                })}
```

Comment:
I don't like the negative margin here, but I couldn't figure out another answer in reasonable time. If anyone has a better pattern in mind, let me know.